zebrad/src/components/tracing/component.rs

```diff
@@ -31,17 +31,29 @@ impl Tracing {
             .with_filter_reloading();
         let filter_handle = builder.reload_handle();
 
-        let subscriber = builder.finish().with(ErrorLayer::default());
-
-        let flamegrapher = if let Some(path) = flame_root {
+        let (flamelayer, flamegrapher) = if let Some(path) = flame_root {
             let (flamelayer, flamegrapher) = flame::layer(path);
-            subscriber.with(flamelayer).init();
-            Some(flamegrapher)
+            (Some(flamelayer), Some(flamegrapher))
+        } else {
+            (None, None)
+        };
+
+        let journaldlayer = if config.journald {
+            let layer = tracing_journald::layer()
+                .map_err(|e| FrameworkErrorKind::ComponentError.context(e))?;
+            Some(layer)
         } else {
-            subscriber.init();
             None
         };
 
+        let subscriber = builder.finish().with(ErrorLayer::default());
+        match (flamelayer, journaldlayer) {
+            (None, None) => subscriber.init(),
+            (Some(layer1), None) => subscriber.with(layer1).init(),
+            (None, Some(layer2)) => subscriber.with(layer2).init(),
+            (Some(layer1), Some(layer2)) => subscriber.with(layer1).with(layer2).init(),
+        };
```

Comment:
This required because `with` (actually [with_subscriber](https://docs.rs/tracing-subscriber/0.2.17/tracing_subscriber/layer/trait.Layer.html#method.with_subscriber)) return `Layered<L, I, S = I>`. As result, we can not use `mut` variable because the type will be changed each time. We can not use `Box<dyn T>` either because `with_subscriber` requires `Sized`.